### PR TITLE
Calculation functions

### PR DESCRIPTION
**Pull request by donacrisante**

### CODE REVIEW

File: components/EntryForm.js [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
+    } else if 
+      (transport === "train") {
+        return calculator[6].train(km);
+    } else if 
+      (transport === "bicycle") {
+        return calculator[7].bicycle(km);
+    } else {
+      console.log("Please select a transport");
+    }
+  }
```
Comment (moonwave99) [inline components/EntryForm.js:40]: I would normalise the calculators as a plain object:

```js
export const calculator = {
  petrolCar: (km) => {
    const petrolEmissionInKg = 0.189;
    return Math.round(petrolEmissionInKg * km * 100) / 100;
  },
  // average value of 7l/100km -- average emission of 2.65kg/l
  dieselCar: (km) => {
    const dieselEmissionInKg = 0.1855;
    return Math.round(dieselEmissionInKg * km * 100) / 100;
  },
  // average value of 4l/100km -- average emission of 2.37kg/l
  hybridCar: (km) => {
    const hybridEmissionInKg = 0.0948;
    return Math.round(hybridEmissionInKg * km * 100) / 100;
  },
  // average value of 18kWh/100km -- average emission of 0,434kg/kWh
  electricStrommixCar: (km) => {
    const strommixEmissionInKg = 0.07812;
    return Math.round(strommixEmissionInKg * km * 100) / 100;
  },
  electricEcoCar: (km) => {
    const ecoEmissionInKg = 0.0;
    return ecoEmissionInKg * km;
  },
  plane: (km) => {
    const planeEmissionInKg = 0.23;
    return Math.round(planeEmissionInKg * km * 100) / 100;
  },
  // average value for long-distance traffic
  train: (km) => {
    const trainEmissionInKg = 0.032;
    return Math.round(trainEmissionInKg * km * 100) / 100;
  },
  bicycle: (km) => {
    const bicycleEmissionInKg = 0.0;
    return bicycleEmissionInKg * km;
  },
};
```

Then you'll need just a single `<select>`, with all the options:

```html
<select>
  <option value="petrolCar">Car - Petrol</option>
  <option value="dieselCar">Car - Diesel</option>
  ...
  <option value="plane">Plane</option>
  ...
</select>
```

(you can still generate it via `map()` - good!)

At the end in `handleCalculateCo2` you drop the `fuel` parameter (and from the state as well):

```js
function handleCalculateCo2(transport, km) {
  return calculator[transport](km);
}
```

Comment (donacrisante) [reply components/EntryForm.js:40]: Hi Diego, thanks a lot for your feedback! I will change my code as you recommended.

File: library/calculator.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return Math.round((hybridEmissionInKg * km) * 100) / 100;
+    }},
+    // average value of 18kWh/100km -- average emission of 0,434kg/kWh
+    { electricStrommix: function calculationElectricStrommix(km) {
+        const strommixEmissionInKg = 0.07812;
+        return Math.round((strommixEmissionInKg * km) * 100) / 100;
+    }},
+    { electricEco: function calculationElectricEco(km) {
+        const ecoEmissionInKg = 0.0;
+        return ecoEmissionInKg * km;
```
Comment (Kheirah) [inline library/calculator.js:25]:
```suggestion
    "electric-strommix": function calculationElectricStrommix(km) {
        const strommixEmissionInKg = 0.07812;
        return Math.round((strommixEmissionInKg * km) * 100) / 100;
    },
    "electric-renewable": function calculationElectricEco(km) {
        const ecoEmissionInKg = 0.0;
        return ecoEmissionInKg * km;
```
If you rename your keys to match the keys in your `EntryForm` you'll be able to keep your two `select` elements and reduce your code in the `handleCalculateCo2` function down to two lines.

File: components/EntryForm.js [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+        return calculator[5].plane(km);
+    } else if 
+      (transport === "train") {
+        return calculator[6].train(km);
+    } else if 
+      (transport === "bicycle") {
+        return calculator[7].bicycle(km);
+    } else {
+      console.log("Please select a transport");
+    }
```
[the remark this answers is not in the document]
Comment (donacrisante) [reply components/EntryForm.js]: Hi Ernst, thank you for your suggestions. I have change my code and it still works ;)

**Review by Kheirah — APPROVED** [review]
Great!

**Pull request merged**